File: BOJ_2023/2979_트럭주차/main.js
// https://www.acmicpc.net/problem/2979

{
  const readline = require('readline')
  const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
  })

  const input = []
  rl.on('line', (line) => {
    input.push(line)
  }).on('close', () => {
    solve()
    process.exit()
  })

  function solve() {
    const [a, b, c] = input.shift().split(' ').map(Number)
    const v = input.map((val) => val.split(' ').map(Number))
    const time = Array(100).fill(0)

    for (let i = 0; i < v.length; i++) {
      for (let j = v[i][0]; j < v[i][1]; j++) time[j]++
    }

    console.log(
      time.reduce((acc, cur) => {
        if (cur === 1) return acc + cur * a
        if (cur === 2) return acc + cur * b
        if (cur === 3) return acc + cur * c
        return acc
      })
    )
  }
}

File: BOJ_2023/2979_트럭주차/main.cpp
// https://www.acmicpc.net/problem/2979

#include <bits/stdc++.h>
using namespace std;
int t[100], a, b, c, x, y;

int main() {
  cin >> a >> b >> c;

  for (int i = 0; i < 3; i++) {
    cin >> x >> y;
    for (; x < y; x++) t[x]++;
  }
  int cost = 0;
  for (int i = 0; i < sizeof(t) / sizeof(int); i++) {
    if (t[i] == 1) cost += a;
    if (t[i] == 2) cost += (b * 2);
    if (t[i] == 3) cost += (c * 3);
  }
  cout << cost << "\n";
  return 0;
}
